src/storm/storage/expressions/UnaryNumericalFunctionExpression.cpp

```diff
@@ -96,6 +108,11 @@ std::shared_ptr<BaseExpression const> UnaryNumericalFunctionExpression::simplify
                 case OperatorType::Floor:
                 case OperatorType::Ceil:
                     break;
+                case OperatorType::Cos:
+                case OperatorType::Sin:
+                    // The operands expect a real value. Raise an exception
+                    STORM_LOG_THROW(false, storm::exceptions::InvalidOperationException, "Cannot apply cos or sin to an integer value.");
```

Comment:
My understanding of the JANI-spec is that e.g. `sin(10)` would be valid in JANI as int is also a valid numeric type.